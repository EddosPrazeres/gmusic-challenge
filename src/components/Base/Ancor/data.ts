export interface IAncor { 
  title: string;
  target?: string;
  link?: string;
  onClick?: any
}

export interface IAncorLayout extends IAncor {  
  
}