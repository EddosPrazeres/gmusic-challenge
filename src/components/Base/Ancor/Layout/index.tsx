import * as React from "react";
import { Text } from "../../../../components";
import { IAncorLayout } from "../data";
import './styles.scss'

export function Ancor({ 
  title,
  target="_blank",
  link,
  onClick,
  ...props 
}: IAncorLayout) {
  return (
    <a 
      className="Ancor"
      target={target}
      onClick={onClick}
      href={link}>
        <Text title={title} size="smaller" />
    </a>
  )
}