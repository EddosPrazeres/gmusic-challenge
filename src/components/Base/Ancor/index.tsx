import * as React from 'react';
import { IAncor } from './data';
import { Ancor as Layout } from './Layout';

export default function Ancor({...props}: IAncor) {
  return (
    <Layout {...props} />
  )
}

  