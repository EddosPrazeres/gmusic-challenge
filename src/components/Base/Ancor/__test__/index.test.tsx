import React from 'react'
import { mount, shallow } from 'enzyme'
import Ancor from '../index'

const createComponent = (props, useDefault = false) => {
  const componentProps = {    
    ...props,
  }

  const wrapper = useDefault ? mount(<Ancor  {...componentProps} />) : shallow(<Ancor     {...componentProps} />)
  const unmount = () => wrapper.unmount()

  return [wrapper, unmount]
}

describe('Ancor', () => {
  it('should match snapshot', () => {
    const [wrapper, unmount] = createComponent({})

    expect(wrapper).toMatchSnapshot()
    unmount()
  })
})
