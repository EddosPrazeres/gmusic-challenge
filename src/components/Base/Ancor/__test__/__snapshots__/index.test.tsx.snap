// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ancor should match snapshot 1`] = `ShallowWrapper {}`;
